clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -111,6 +117,10 @@ public ByteArraySerde() {
             return (Serde<T>) Long();
         }
 
+        if (Float.class.isAssignableFrom(type)) {
```

Comment:
I have a question: Would `type instanceof Float` be better? Or, you decided to use `Class#isAssignableFrom` method for consistency only?